### PR TITLE
DT-876 Can filter by active on a certain date

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/data/filters/OffenderFilter.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.digital.delius.data.filters;
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/data/filters/OffenderFilter.java:1]: Above spilt between the "model" and the JPA predicate stuff (below this file)

File: src/main/java/uk/gov/justice/digital/delius/jpa/standard/entity/Offender.java [hunk 2]
```diff
@@ -90,31 +90,31 @@
     @Column(name = "ROW_VERSION")
     private Long rowVersion;
 
-    @ManyToOne
+    @ManyToOne(fetch = FetchType.LAZY)
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/jpa/standard/entity/Offender.java:93]: All these were eager cause several selects for each retrieve for an offender even when these fields were never returned

File: src/testIntegration/java/uk/gov/justice/digital/delius/controller/api/ConvictionAPITest.java [hunk 3]
```diff
@@ -54,7 +54,7 @@ public void canGetConvictionsByCrn() {
             .body()
             .as(Conviction[].class);
 
-        assertThat(convictions).extracting("convictionId").containsExactlyInAnyOrder(2500297061L, 2500295345L, 2500295343L);
+        assertThat(convictions).extracting("convictionId").containsExactlyInAnyOrder(2500297061L, 2500295345L, 2500295343L, 10002L, 10003L);
```
Comment (andymarke) [inline src/testIntegration/java/uk/gov/justice/digital/delius/controller/api/ConvictionAPITest.java:57]: added a few more events to this offender

File: src/testIntegration/java/uk/gov/justice/digital/delius/controller/secure/CustodyUpdateBookingNumberAPITest.java [hunk 4]
```diff
@@ -54,7 +54,7 @@ public void updateBookingNumber() {
         final var custody = given()
                 .auth().oauth2(token)
                 .contentType("application/json")
-                .body(createUpdateCustodyBookingNumber("V74999", LocalDate.of(2019, 9, 5)))
+                .body(createUpdateCustodyBookingNumber("V74999", LocalDate.of(2019, 9, 21)))
```
Comment (andymarke) [inline src/testIntegration/java/uk/gov/justice/digital/delius/controller/secure/CustodyUpdateBookingNumberAPITest.java:57]: changed date for other tests

File: src/main/java/uk/gov/justice/digital/delius/jpa/standard/entity/Offender.java (hunk 2 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/digital/delius/jpa/standard/entity/Offender.java:93]: wow! Didn't realise these were all being EAGER loaded!

**Pull request merged**